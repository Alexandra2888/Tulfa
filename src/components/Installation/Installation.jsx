import React from "react";
import { Swiper, SwiperSlide } from "swiper/react";
import { ChevronUp, ChevronDown } from "lucide-react";
import "swiper/css";
import Title from "../../atoms/Title/Title";
import Description from "../../atoms/Description/Description";
import Button from "../../atoms/Button/Button";


const installationData = {
  slides: [
    {
      id: 1,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731231754/tulfa/Installation/happy-couple-carrying-furniture-into-their-new-apa-2023-11-27-05-33-12-utc_vgzpf3.jpg",
      title: "Living Room Setup",
      alt: "Modern living room furniture installation",
    },
    {
      id: 2,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731231905/tulfa/Installation/male-carpenter-making-wooden-designer-furniture-fo-2023-11-27-05-07-47-utc_f6jcu6.jpg",
      title: "Bedroom Collection",
      alt: "Contemporary bedroom set",
    },
    {
      id: 3,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731231949/tulfa/Installation/male-carpenter-making-wooden-designer-furniture-fo-2023-11-27-05-12-13-utc_ylckp8.jpg",
      title: "Office Space",
      alt: "Professional office furniture layout",
    },
    {
      id: 4,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731231982/tulfa/Installation/male-carpenter-making-wooden-designer-furniture-fo-2023-11-27-05-12-45-utc_yb1znh.jpg",
      title: "Office Space",
      alt: "Professional office furniture layout",
    },
    {
      id: 5,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731232010/tulfa/Installation/young-man-in-a-furniture-factory-2023-11-27-05-26-18-utc_rlvcx0.jpg",
      title: "Office Space",
      alt: "Professional office furniture layout",
    },
    {
      id: 6,
      image:
        "https://res.cloudinary.com/dnpjmrdik/image/upload/v1731231740/tulfa/Installation/closeup-professional-carpenter-hold-electric-circu-2024-01-04-21-15-19-utc_bf78o6.jpg",
      title: "Office Space",
      alt: "Professional office furniture layout",
    },
  ],
};

const Installation = () => {
  const [swiper, setSwiper] = React.useState(null);
  const { slides } = installationData;
  const [currentSlide, setCurrentSlide] = React.useState(0);

  const nextSlide = () => {
    if (swiper) {
      swiper.slideNext();
    }
  };

  const prevSlide = () => {
    if (swiper) {
      swiper.slidePrev();
    }
  };

  return (
    <section className="relative bg-white">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 flex flex-col items-center">
        {/* Title Section */}
        <Title text="Lorem Ipsum" />
        {/* Carousel Section */}
        <div className="w-[80%] xl:w-full relative mb-12">
          <div className="aspect-[4/3] w-full relative overflow-hidden rounded-lg">
            <Swiper
              direction="vertical"
              onSwiper={setSwiper}
              speed={500}
              onSlideChange={(swiper) => setCurrentSlide(swiper.activeIndex)}
              className="h-full"
              style={{ height: "100%" }}
              loop={true}
            >
              {slides.map((slide) => (
                <SwiperSlide key={slide.id} className="h-full">
                  <img
                    src={slide.image}
                    alt={slide.alt}
                    className="w-full h-full object-cover"
                  />
                </SwiperSlide>
              ))}
            </Swiper>

            {/* Navigation Buttons */}
            <div className="absolute right-4 top-1/2 -translate-y-1/2 flex flex-col gap-2 z-10">
              <Button
                onClick={prevSlide}
                className="border-2 border-white rounded-full w-10 h-10 md:w-12 md:h-12 flex items-center justify-center transition-all hover:bg-white/20 active:bg-white/30 touch-manipulation"
                aria-label="Previous slide"
              >
                <ChevronUp className="w-5 h-5 md:w-6 md:h-6 text-white" />
              </Button>
              <Button
                onClick={nextSlide}
                className="border-2 border-white rounded-full w-10 h-10 md:w-12 md:h-12 flex items-center justify-center transition-all hover:bg-white/20 active:bg-white/30 touch-manipulation"
                aria-label="Next slide"
              >
                <ChevronDown className="w-5 h-5 md:w-6 md:h-6 text-white" />
              </Button>
            </div>

            {/* Slide Counter */}
            <div className="absolute bottom-4 left-1/2 -translate-x-1/2 bg-gray-900/75 text-white px-4 py-2 rounded-full text-sm z-10">
              {currentSlide + 1} / {slides.length}
            </div>
          </div>
        </div>

        {/* Description Section */}
        <div className="w-full max-w-6xl mx-auto mb-12">
         <Description text="Give your customers a clear view of how your furniture fits into their space with precise dimensions and scale indicators."
/>
        </div>
      </div>
    </section>
  );
};

export default Installation;
