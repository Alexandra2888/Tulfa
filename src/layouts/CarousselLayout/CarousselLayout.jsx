import React from "react";
import { Swiper, SwiperSlide } from "swiper/react";
import { ChevronUp, ChevronDown } from "lucide-react";
import "swiper/css";
import Title from "../../atoms/Title/Title";
import Description from "../../atoms/Description/Description";
import Button from "../../atoms/Button/Button";

const CarouselLayout = ({ title, description, slides }) => {
  const [swiper, setSwiper] = React.useState(null);
  const [currentSlide, setCurrentSlide] = React.useState(0);

  const nextSlide = () => swiper?.slideNext();
  const prevSlide = () => swiper?.slidePrev();

  return (
      <section className="w-full bg-white">
        {/* Title Section in Container */}
        <div className="container mx-auto px-4 max-w-7xl">
          <div className="w-full py-12 lg:py-24 flex justify-center items-center">
            <Title text={title} className="text-4xl lg:text-5xl font-bold text-center" />
          </div>
        </div>

        {/* Full-width Carousel Section */}
        <div className="w-full relative">
          <div className="w-full aspect-[16/9] relative">
            <Swiper
                direction="vertical"
                onSwiper={setSwiper}
                speed={500}
                onSlideChange={(swiper) => setCurrentSlide(swiper.activeIndex)}
                className="h-full w-full"
                loop={true}
            >
              {slides.map((slide) => (
                  <SwiperSlide key={slide.id} className="w-full h-full">
                    <div className="relative w-full h-full">
                      <img
                          src={slide.image}
                          alt={slide.alt}
                          className="w-full h-full object-cover"
                      />
                      <div className="absolute inset-0 bg-black/10" />
                    </div>
                  </SwiperSlide>
              ))}
            </Swiper>

            {/* Navigation Buttons */}
            <div className="absolute right-6 lg:right-12 top-1/2 -translate-y-1/2 flex flex-col gap-3 z-10">
              <Button
                  onClick={prevSlide}
                  className="bg-white/10 backdrop-blur-sm border border-white/20 rounded-full w-12 h-12 flex items-center justify-center transition-all hover:bg-white/30 active:bg-white/40"
                  aria-label="Previous slide"
              >
                <ChevronUp className="w-6 h-6 text-white" />
              </Button>
              <Button
                  onClick={nextSlide}
                  className="bg-white/10 backdrop-blur-sm border border-white/20 rounded-full w-12 h-12 flex items-center justify-center transition-all hover:bg-white/30 active:bg-white/40"
                  aria-label="Next slide"
              >
                <ChevronDown className="w-6 h-6 text-white" />
              </Button>
            </div>

            {/* Slide Counter */}
            <div className="absolute bottom-6 left-6 lg:left-12 bg-white/10 backdrop-blur-sm border border-white/20 text-white px-4 py-2 rounded-full text-sm z-10">
              {currentSlide + 1} / {slides.length}
            </div>
          </div>
        </div>

        {/* Description Section in Container */}
        <div className="container mx-auto px-4 max-w-7xl">
          <div className="w-full mx-auto px-4 xl:px-0">
            <Description
                text="Give your customers a clear view of how your furniture fits into their space with precise dimensions and scale indicators."
            />
          </div>
        </div>
      </section>
  );
};

export default CarouselLayout;