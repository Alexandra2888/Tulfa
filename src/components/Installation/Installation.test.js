// Installation.test.jsx
import { render, screen } from "@testing-library/react";
import Installation from "./Installation";
import CarousselLayout from "../../layouts/CarousselLayout/CarousselLayout";

// Mock the dependencies
jest.mock("../../layouts/CarousselLayout/CarousselLayout", () => {
  return jest.fn(({ title, description, slides }) => (
    <div data-testid="carousel-layout">
      <h2 data-testid="carousel-title">{title}</h2>
      <p data-testid="carousel-description">{description}</p>
      <div data-testid="carousel-slides">
        {slides.map((slide, index) => (
          <div key={index} data-testid={`slide-${index}`}>
            {slide.title}
          </div>
        ))}
      </div>
    </div>
  ));
});

jest.mock("../../data/data", () => ({
  installationData: {
    slides: [
      { title: "Slide 1", image: "/image1.jpg" },
      { title: "Slide 2", image: "/image2.jpg" },
    ],
  },
}));

describe("Installation Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing", () => {
    render(<Installation />);
    expect(screen.getByTestId("carousel-layout")).toBeInTheDocument();
  });

  it("passes correct props to CarouselLayout", () => {
    render(<Installation />);

    // Check if title is correct
    expect(screen.getByTestId("carousel-title")).toHaveTextContent(
      "Lorem Ipsum"
    );

    // Check if description is correct
    expect(screen.getByTestId("carousel-description")).toHaveTextContent(
      "Give your customers a clear view of how your furniture fits into their space with precise dimensions and scale indicators."
    );

    // Check if slides are passed correctly
    expect(screen.getByTestId("slide-0")).toHaveTextContent("Slide 1");
    expect(screen.getByTestId("slide-1")).toHaveTextContent("Slide 2");
  });

  it("calls CarouselLayout with correct number of slides", () => {
    render(<Installation />);
    const slides = screen.getAllByTestId(/^slide-/);
    expect(slides).toHaveLength(2);
  });

  it("matches snapshot", () => {
    const { container } = render(<Installation />);
    expect(container).toMatchSnapshot();
  });
});
