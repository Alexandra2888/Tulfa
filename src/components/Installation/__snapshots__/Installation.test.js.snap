// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Installation Component matches snapshot 1`] = `
<div>
  <div
    data-testid="carousel-layout"
  >
    <h2
      data-testid="carousel-title"
    >
      Lorem Ipsum
    </h2>
    <p
      data-testid="carousel-description"
    >
      Give your customers a clear view of how your furniture fits into their space with precise dimensions and scale indicators.
    </p>
    <div
      data-testid="carousel-slides"
    >
      <div
        data-testid="slide-0"
      >
        Slide 1
      </div>
      <div
        data-testid="slide-1"
      >
        Slide 2
      </div>
    </div>
  </div>
</div>
`;
